### PR TITLE
Integration of Profile Details

### PR DESCRIPTION
**Pull request by jb162**
### What this PR does (required):
- addresses #62 
- renders ProfileDetails dynamically via URL (e.g. /profile-details/61affa655479b0ee3cc36331 shows profile details about the user with an id of 61affa655479b0ee3cc36331)
- BookingsForm is populated with the availability of the user of interest
- submitting a request creates a notification for that user that a service request has been placed

### Screenshots / Videos (front-end only):
image

### Any information needed to test this feature (required):
- run server and app
- login with your account
- copy and paste your user id into client/mocks/mockId.tsx
- navigate to /profile-details/<your-user-id-here>
- the profile details card should populate with the information in your profile (can change via EditProfile)
- make an appointment via BookingsForm (make sure your profile has availableTime property in database; the code that changes the profile model to include this property is in FE/IN Availability Page for reference)
image
- check mongoDB to see if new notification has been created for service request

### Any issues with the current functionality (optional):
- had to change the date input component to a select component
- need to add validation
- need to add hourly rate and rating to profile model
- need to finish work on the integration of availability page (#84) since this PR is dependent on it

### CODE REVIEW

File: client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx
Comment (ezeilo-su) [inline client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx:105]: Let's get rid of this unused code block please :smiley:

File: client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx
Comment (ezeilo-su) [inline client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx:150]: Also remove this unused code

**Review by ezeilo-su — COMMENTED** [review]
Looks good! A few minor comments, though :smile:

File: client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx
Comment (ugur-akin) [inline client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx:68]: It's nice to see some form of feedback being communicated with the user, although I suggest we do this through [Snackbars](https://v4.mui.com/components/snackbars/#snackbar) instead (we already have the context implemented in the codebase ([here](https://github.com/hatchways/team-gingerbread/blob/justin-integration-profile-details/client/src/context/useSnackbarContext.tsx)) 🎉) !

File: client/src/helpers/APICalls/createRequestNotification.ts
Comment (ugur-akin) [inline client/src/helpers/APICalls/createRequestNotification.ts:20]: Looks solid 💯 My only suggestion would be to indicate that `recipient` field expects an id by renaming it (e.g. `recipientId` or `recipientProfileId`)

File: client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx
Comment (ugur-akin) [inline client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx:77]:
```suggestion
    const endTime = typeof startDate?.endTime === 'number' ? parseInt(startDate?.endTime) : 24;
```

**Review by ugur-akin — COMMENTED** [review]
Looks good! Just mentioned a couple of minor concerns and highlighted a line that seems to have a typo. We should be able to merge this soon 🙌🏼

File: client/src/App.tsx
Comment (rajivtitus) [inline client/src/App.tsx:37]: 💯

File: client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx
Comment (rajivtitus) [inline client/src/components/EditProfile/EditProfileTab/EditProfileTab.tsx:46]: I think this is self-explanatory, the comment on this line can be excluded :)

File: client/src/pages/ProfileDetails/BookingForm/useStyles.ts
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/BookingForm/useStyles.ts:26]: Most of these styles can be added directly to the Box component above

File: client/src/pages/ProfileDetails/useStyles.ts
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/useStyles.ts:69]: 👍🏽

File: client/src/pages/ProfileDetails/useStyles.ts
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/useStyles.ts:58]: Optional: Consider setting the font size for variants in the theme and using those variants across the app

**Review by rajivtitus — COMMENTED** [review]
Excellent work! 💯 
Just some minor comments that need to be addressed 👍🏽

File: client/src/pages/ProfileDetails/BookingForm/useStyles.ts
Comment (jb162) [reply client/src/pages/ProfileDetails/BookingForm/useStyles.ts:26]: I was getting some spacing issues when adding directly, so I kept them in the useStyles file

File: client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx
Comment (rajivtitus) [reply client/src/pages/ProfileDetails/BookingForm/BookingForm.tsx:68]: @jb162 Please consider using the Snackbar here 👍🏽

File: client/src/pages/ProfileDetails/ProfileDetails.tsx
Comment (rajivtitus) [inline client/src/pages/ProfileDetails/ProfileDetails.tsx:25]: Please remove commented code or add a TODO comment

**Review by rajivtitus — APPROVED** [review]
LGTM! Please address pending comments before merging 👍🏽

File: client/src/pages/ProfileDetails/ProfileDetails.tsx
Comment (jb162) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx:25]: I'm having an issue with the useParams hook not working. It was previously working but now is unable to get the id parameter from the URL. Could you take a look at this?

Comment (rajivtitus) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx:25]: @jb162 Are we getting any error from the above line of code? Also, did we change the profile details route?

Comment (jb162) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx:25]: From Line 25, I am getting "Unhandled Rejection (TypeError): Cannot read properties of undefined (reading 'profile')". I tried to log the parameters but the useParams hook is not working at all. When logging the params variable that is set equal to useParams(), I receive an object but it does not include the id parameter. The profile details route has not been changed and is set to "/profile-details/:id".

Comment (rajivtitus) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx:32]: @jb162 The useParams hook is generic, we need to tell typescript which params we are using by specifying the value of the generic

```
const { sumParams } = useParams<{ id?: string }>();
```

Comment (jb162) [reply client/src/pages/ProfileDetails/ProfileDetails.tsx:32]: I was still having this issue even after implementing the above line. For some reason, useParams() returned an empty object every time. The only way I was able to fix it was by switching from a ProtectedRoute to a normal Route component.

**Comment by jb162** [comment]
@rajivtitus 

I have addressed the pending comments and was able to add all the features I wanted. The only issue is that the ProfileDetails page only works if it is a component of a Route rather than a ProtectedRoute. The latter causes an issue with the useParams() hook and prevents me from accessing the id of the currently displayed profile. How should I proceed?

**Comment by rajivtitus** [comment]
> @rajivtitus
> 
> I have addressed the pending comments and was able to add all the features I wanted. The only issue is that the ProfileDetails page only works if it is a component of a Route rather than a ProtectedRoute. The latter causes an issue with the useParams() hook and prevents me from accessing the id of the currently displayed profile. How should I proceed?

For the time being you can use a regular Route component (eventually we will have parts of the Dashboard/Listings page and the Profile Details page as unprotected routes)

**Pull request merged**
